src/layout/strategy/greedy.js

```diff
@@ -0,0 +1,80 @@
+import d3 from 'd3';
+import {rebindAll} from '../../util/rebind';
+import collision from './collision';
+import containerUtils from './container';
+
+export default function() {
+
+    var container = containerUtils();
+    var collider = collision();
+
+    var strategy = function(data) {
+        var builtPoints = [];
+
+        // Use forEach over map so the array can be built one at a time.
+        data.forEach(function(point) {
+            var allPointPlacements = getAllPlacements(point);
+            var scoredPlacements = allPointPlacements.map(function(unscoredPlacement) {
+                return checkPlacement(builtPoints, unscoredPlacement);
+            });
+
+            var bestScore = Infinity;
```

Comment:
We have similar logic implemented using map-reduce elsewhere:

https://github.com/ScottLogic/d3fc/blob/master/src/util/snap.js#L19

(which is just showing off ;-) )

I think this is a good candidate for a utility function, that takes an array of items and a scoring function.